src/common/quantile.cc

```diff
@@ -371,7 +369,8 @@ auto AddCategories(std::set<float> const &categories, HistogramCuts *cuts) {
     InvalidCategory();
   }
   auto &cut_values = cuts->cut_values_.HostVector();
-  auto max_cat = *std::max_element(categories.cbegin(), categories.cend());
```

Comment:
@trivialfis I think this is a legit bug. If `categories` is empty, `std::max_element` returns `categories.cend`, which leads to undefined behavior when accessed.

Presumably with row-split `categories` would never be empty, but with column-split it's pretty common.

Reply:
Done.